.travis_scripts/build-release.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -e
+set -x
+
+# JAVA_HOME isn't set on OSX for some reason, so manually set it
+if [ -z "$JAVA_HOME" ]; then
+    export JAVA_HOME=$(/usr/libexec/java_home)
+fi
+
+cargo b --release
+
+# copy out just the library to upload
+mkdir release_artifacts
+shopt -s extglob
```

Comment:
If you wouldn't mind commenting about why this is needed, that'd be great.

Reply:
In the deploy we need to grab the `.so` or the `.dylib`, whichever has been built for the current os. I don't want to grab all the files or all the top level files in `release` because there's junk, and globbing to `ironoxide_java.*` would get the `.d` file too. None of the commands can fail (so I can't do `release/libironoxide_java.{so,dylib}` or `-e` will bail. Without globbing (just using multiple files listed in the deploy) it is all inclusive, so it'll fail if only one is there. I looked into using `cargo --out-dir` to get only the libs without intermediate files but it's not in stable.

Basically after trying a lot of junk the solution was to use an `|` glob so if either of them is there (or both for some reason) it'll grab them and only them and put them into a spot that I can grab with the `file` directive on the `deploy` stage. `|` globs are a bash option that needs to be enabled. Hence `shopt -s extglob`. If you want that in the file as a comment, how much of it should be there?

Reply:
I would extend the comment above to say 

> Copy out just the library to upload. In order do just grab the .dylib or .so we just use extglob. We did this instead of 2 cp lines so we could leave `-e` enabled and it would succeed as long as we have at least one of them